### PR TITLE
#81 - Support Basic authentication

### LINKED ISSUE #81: Support Basic authentication
It would be good to support `Basic` authentication for Gem to install passing credentials in the url. To do this, you need to modify the method `GemApiKeyIdentities#user`

### PR DESCRIPTION
**Pull request by genryxy**
Closes #81 
Support Basic authentication. Add test to verify it

### CODE REVIEW

File: src/main/java/com/artipie/gem/GemApiKeyIdentities.java [hunk 1]
```diff
@@ -55,6 +55,18 @@ public GemApiKeyIdentities(final Authentication auth) {
...
+                    if (str.startsWith(basic)) {
+                        res = str.substring(basic.length());
+                    } else {
+                        res = str;
+                    }
+                    return res;
+                }
+            )
             .map(Base64Decoded::new)
             .map(dec -> dec.toString().split(":"))
```
Comment (olegmoz) [inline src/main/java/com/artipie/gem/GemApiKeyIdentities.java:71]: @genryxy I'd suggest to use parsing of header value from `Authorized.Basic` class from HTTP module

File: src/test/java/com/artipie/gem/AuthTest.java [hunk 2]
```diff
@@ -132,4 +139,50 @@ public void notAllowedToInstallsUsersAreRejected() throws IOException {
...
+                Arrays.asList(
+                    new RsHasStatus(RsStatus.UNAUTHORIZED),
+                    new RsHasHeaders(new Header("WWW-Authenticate", "Basic"))
+                )
+            )
+        );
+    }
+
+    @Test
+    public void returnsOk() throws IOException {
```
Comment (olegmoz) [inline src/test/java/com/artipie/gem/AuthTest.java:157]: @genryxy please describe conditions for successful auth in this test's name

File: src/test/java/com/artipie/gem/AuthTest.java [hunk 3]
```diff
@@ -132,4 +139,50 @@ public void notAllowedToInstallsUsersAreRejected() throws IOException {
...
+        MatcherAssert.assertThat(
+            AuthTest.post(true),
+            new RsHasStatus(RsStatus.OK)
+        );
+    }
+
+    private static Response post(final boolean authorized) throws IOException {
```
Comment (olegmoz) [inline src/test/java/com/artipie/gem/AuthTest.java:164]: @genryxy I think it should be stated in method name that `POST` is done with `Basic` auth header. We test another auth method in this class as well

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see some comments

**Comment by genryxy** [comment]
@olegmoz thanks, I used `Authorization` class after checking that header value starts with `Basic` because in some cases header value can be represented only by token. This required a version update for http module therefore todos were added.

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, merging

**Pull request merged**